### PR TITLE
Rewrite lib/gengi for more functionality & tests

### PR DESCRIPTION
**Pull request by benediktvaldez**
Also, updated usage instructions accordingly

Also, TESTS

### CODE REVIEW

File: lib/args.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import each from 'lodash/each'
```
Comment (koddsson) [inline lib/args.js]: Why not `arr.forEach(item => ???)`?

File: lib/gengi.test.js [hunk 2]
```diff
@@ -1,10 +1,91 @@
 import { expect } from 'chai'
-import gengi from './gengi'
+import Gengi from './gengi'
 
-describe('Gengi utility', () => {
+describe('lib/gengi - unit', () => {
   it('should be a function', () => {
-    expect(gengi).to.be.a('function')
+    expect(Gengi).to.be.a('function')
```
Comment (koddsson) [inline lib/gengi.test.js:6]: I wish this line could be `expect(Gengi).to.be.a('class')` but I'm expecting it to be a drawback from transpiling?

Comment (benediktvaldez) [inline lib/gengi.test.js:6]: Yep, it's technically a function at this point :disappointed:

File: lib/args.js (hunk 1 above)
Comment (benediktvaldez) [inline lib/args.js]: Fair point - fixed

Comment (koddsson) [inline lib/args.js]: Could we then do it everywhere and remove the `lodash` dependency?

Comment (benediktvaldez) [inline lib/args.js]: Done :smile_cat:

**Comment by koddsson** [comment]
LGTM!

**Pull request merged**